Off-Top-App/off-top-flutter
# OT-MigrationToFlutter1_17 | Changed entire codebase to follow recommended specifications of Flutter 1.17

**What I Did:**

- I changed code where needed to follow the recommended specifications for Flutter 1.17. The underlying functions of the code should NOT have changed at all. Majority of the changes were specifying type annotations, removing any `new` keywords, and using a ` ' ` instead of `"`.

- *Won't Fix* For meter.dart, line 26 `final dynamic incomingData = json.decode(onData);` and line 27  `isOnTopic = incomingData['focus_score'];`cannot be properly migrated to Flutter 1.17 without breaking functionality to my knowledge. As of now, the code will still run without problems, but unsure this will be the case if Flutter updates again in the future.

- I also cleaned up the spacing formatting of the code following Flutter 1.17 specifications.

**How To Test:**

- Prerequisites: You will need the latest flutter build (Flutter 1.17.3). Run command: `flutter upgrade` in a terminal to get this build. 

- Please test this branch very carefully! I do not understand the entire intended functionality right now and did the best to my knowledge. 

- You will need Zookeeper, Kafka, Spring-boot, and the MySQL servers running.

- Test every functionality of the app and it's connection to the backend. Please make sure I did not break anything by migrating to Flutter 1.17.
